moveit_ros/moveit_servo/src/servo_parameters.cpp

```diff
@@ -77,74 +78,259 @@ ServoParameters::setParametersCallback(std::vector<rclcpp::Parameter> parameters
   return result;
 }
 
-ServoParameters::SharedConstPtr ServoParameters::makeServoParameters(const rclcpp::Node::SharedPtr& node,
-                                                                     const rclcpp::Logger& logger,
-                                                                     std::string ns, /* = "moveit_servo"*/
-                                                                     bool dynamic_parameters /* = true */)
+void ServoParameters::declare(const std::string& ns,
+                              const rclcpp::node_interfaces::NodeParametersInterface::SharedPtr& node_parameters,
+                              std::shared_ptr<ServoParameters> parameters)
```

Comment:
Similar to the get function, this shouldn't need parameters shared ptr parameter.  Also, as declaring does not set any values (it only describes the parameter interface to the ros system) it shouldn't set values in a parameters object.  Internally you should just create a temporary one for the default values.  Something like this should work:

```suggestion
void ServoParameters::declare(const std::string& ns,
                              const rclcpp::node_interfaces::NodeParametersInterface::SharedPtr& node_parameters)
{
  auto parameters = ServoParameters{ns};  // for populating default values
```

Lastly, you should be able to get rid of the logger out of the constructor and just put one in this file's scope similar to what is done in other source files in moveit_servo.